CenterForOpenScience/waterbutler
# [SVCS-353]Look for Dataverse renamed files on upload

Dataverse 'ingests' certain file types. These file types get renamed.
In upload when Waterbutler tries to find the correct metadata to return,
it will 500 since it  was not looking for the renamed file.







## Ticket
https://openscience.atlassian.net/browse/SVCS-353


## Purpose
Uploading some file types to dataverse would cause a 500 error to be thrown in waterbutler. The file would still upload, but would have a new extension.



## Changes
Allow waterbutler to return the correct metadata based on the current file name, as well as a new property called `file.original_name`. This new property will parse DV raw metadata to figure out what the original file type was. This way WB can return the correct metadata and won't throw a 500.
added tests for new functions.




## Side effects
Currently DV just deletes old files if you try to upload a conflicting file name. (unsure if this is correct functionality or not)
Because DV changes the file names when uploading certain file types (csv, xlsx, and others), it wont be able to delete the original because it will look like it does not exist. Instead DV uses their own renaming scheme <conflicting-file-1>, <conflicting-file-2< ... 
Because of this, when a file that DV renames gets uploaded, if it got renamed with a -1, or a -2 etc, WB won't look for that. instead it will return the metadata of the non-renamed version.
This shouldn't be much of a problem, as it is really only used for updating fangorn. A page refresh negates this error

I had thought of a few ways to negate this, but decided against it as it would add quite a few more DV api calls to the process. If it is determined in CR that we want to do that, then it can be implemented



## QA Notes
The effect file types are as follows: .csv, .RData, .sav, .dta, .por, .xlsx
If you upload one of these files types to DV, it will be converted into a `.tab` file via their `ingestion` process.

Steps: 
* there are some sample file types on the JIRA ticket to use.
* do not batch upload all the files (DV will return a 400 since it will get stuck on one of the ingestion operations)
* One by one, upload each file
* it should complete successfully and show up in fangorn as <name.tab>.
* clicking on it should go to the right file.

Steps for 'side-effects' notes:
* upload a file for the second time
* it will look like 2 of the same file are on the page
* refresh the page and one of them should show up with a -1 



## Deployment Notes
